File: privactivity-store-rest/src/main/java/net/privactivity/store/unified/ActivityFilterRequest.java
package net.privactivity.store.unified;

import net.privactivity.store.usecase.findActivities.SortDirection;

public record ActivityFilterRequest(
        Boolean includeWaypoints,
        Integer minDistance,
        Integer maxDistance,
        Integer minAscent,
        Integer maxAscent,
        Double lat,
        Double lon,
        Integer radius,
        String title,
        String sortBy,
        SortDirection sortDirection,
        String tileId) {
}

File: angular/src/app/activity-filter-request.ts
export interface ActivityFilterRequest {
    includeWaypoints?: boolean;
    minDistance?: number;
    maxDistance?: number;
    minAscent?: number;
    maxAscent?: number;
    tileId?: string;
    lat?: number;
    lon?: number;
    radius?: number;
    title?: string;
    sortBy?: string;
    sortDirection?: string;
}
